index.js

```diff
@@ -150,7 +155,7 @@ function filter(patterns, opts) {
 
     fp = utils.unixify(fp, opts);
     while (i < len) {
-      var fn = matcher(patterns[i++], opts);
+      var fn = patterns[i++];
       if (!fn(fp)) {
         res = false;
```

Comment:
These are great changes. nice work. 

@es128 any objections to merging? I was going to do a 2.4.0 patch release with housekeeping stuff, like changing to eslint, but maybe we just merge this and do 2.4.0. then I can follow up with a 2.5.0. sound ok?